### ISSUE TITLE
Error connecting to DocumentDB over the port-forwarded connection

### ISSUE THREAD

**Issue by project-administrator** [Kaginari/terraform-provider-mongodb#17]
Hi, 

I'm trying to connect to the DocumentDB which does not have the public IP. Therefore, I've set the port-forwarding on localhost and I'm able to connect with a mongo-cli: 
```
$ mongo --ssl --tlsAllowInvalidHostnames --host 127.0.0.1:27017 --sslCAFile rds-combined-ca-bundle.pem  
2021-07-15T18:39:24.430+0300 W  CONTROL  [main] Option: ssl is deprecated. Please use tls instead.
2021-07-15T18:39:24.430+0300 W  CONTROL  [main] Option: sslCAFile is deprecated. Please use tlsCAFile instead.
MongoDB shell version v4.2.14
connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
2021-07-15T18:39:24.960+0300 W  NETWORK  [js] The server certificate does not match the host name. Hostname: 127.0.0.1 does not match SAN(s): xxxxxxxx.yyyyyy.ca-central-1.docdb.amazonaws.com, xxxxxxxxx.cluster-yyyyyyy.ca-central-1.docdb.amazonaws.com, xxxxxxxxx.cluster-ro-yyyyyyyyy.ca-central-1.docdb.amazonaws.com, CN: xxxxxxx.yyyyyyy.ca-central-1.docdb.amazonaws.com
Implicit session: session { "id" : UUID("some-random-uuid") }
MongoDB server version: 4.0.0
WARNING: shell and server versions do not match
rs0:PRIMARY> 
```
However, trying to create a new user with the following configuration does not work: 
```
terraform {
  required_providers {
    mongodb = {
      source  = "Kaginari/mongodb"
    }
  }
}

provider "mongodb" {
  host = "localhost"
  port = "27017"
  username = local.master_username
  password = random_password.master_password.result
#  username = ""
#  password = ""
  ssl = true
  insecure_skip_verify = true
  certificate = file("${path.module}/files/rds-combined-ca-bundle.pem")
}

resource "mongodb_db_user" "user" {
  auth_database = "admin"
  name = "user1"
  password = "example"
  role {
    role = "readWriteAnyDatabase"
    db =   "some_db"
  }
}
```
The error is the following:
```
│ Error: Error connecting to database : context deadline exceeded 
│ 
│   with module.documentdb.mongodb_db_user.user,
│   on ../../../../documentdb/mongodb.tf line 22, in resource "mongodb_db_user" "user":
│   22: resource "mongodb_db_user" "user" {
```
Adding TF_LOG=trace to the terraform run does not reveal much, provider does not seem to print any debug info. 

Also, trying to sniff the traffic communication with Wireshark between the terraform/mongodb provider and the DocumentDB endpoint is not trivial because it's TLS....

**Comment by ITMonta** [comment]
(using AWS DocumentDB). The issue seems to be entirely within the driver before any commands are sent to the server.
 it seems AWS doesn’t allow auto discover to all the endpoints. Basically, it allows only one cluster endpoint(primary) connection while tunneling from local environment  we need to set direct connect to the primary endpoint.

**Comment by project-administrator** [comment]
Also, probably worth mentioning, that it's a testing single-node DocumentDB cluster. I.e. there is only one single endpoint that exists for this cluster (which I'm trying to connect to).

**Comment by solidnerd** [comment]
Hey @ITMonta  @project-administrator, 
I could figure it out and fix it in the configuration to support this behaviour as well. I did a quick check and it works for me.

```
terraform {
  required_providers {
    mongodb = {
      source = "registry.terraform.io/Kaginari/mongodb"
      version = "9.9.9"
    }
  }
}

provider "mongodb" {
  host = "localhost"
  port = "27017"
  username = "<RANDOM_USERNAME>"
  password =  "<RANDOM_PASSWORD>"
#  username = ""
#  password = ""
  ssl = true
  direct = true
  insecure_skip_verify = true
  certificate = file("${path.module}/files/rds-combined-ca-bundle.pem")
}

resource "mongodb_db_user" "user" {
  auth_database = "<RANDOM_USERNAME>"
  name = "monta"
  password = "monta"
  role {
    role = "readAnyDatabase"
    db =   "admin"
  }
  role {
    role = "readWrite"
    db =   "local"
  }
  role {
    role = "readWrite"
    db =   "monta"
  }

}

```

```
mongodb_db_user.user: Creating...
mongodb_db_user.user: Creation complete after 1s [id=YWRtaW4ubW9udGE=]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

**Comment by ITMonta** [comment]
@solidnerd thank you for the contribution :)

**Comment by ITMonta** [comment]
direct param available in v0.1.2 thanks to  @solidnerd  :+1:
